eth/tracers/api_blocktrace.go

```diff
@@ -325,8 +366,11 @@ func (api *API) getTxResult(env *traceEnv, state *state.StateDB, index int, bloc
 			}
 			env.sMu.Lock()
 			m[keyStr] = wrappedProof
-			if sibling != nil {
-				env.DeletionProofs = append(env.DeletionProofs, sibling)
+			if zktrieTracer.Available() {
+				if isDelete {
+					zktrieTracer.MarkDeletion(key)
+				}
+				env.zkTrieTracer[addrStr].Merge(zktrieTracer)
```

Comment:
Can't we just do `env.zkTrieTracer[addrStr].MarkDeletion(key)`?

Reply:
It cause more nodes (if a node with this key is in fact not involving in any deletion in this block) being included into deletion prof needlessly